debezium-connector-mysql/src/test/java/io/debezium/connector/mysql/BinlogReaderIT.java

```diff
@@ -264,4 +266,92 @@ public void shouldHandleTimestampTimezones() throws Exception {
+
+        SourceRecord sourceRecord = sourceRecords.get(0);
+        Struct value = (Struct) sourceRecord.value();
+        Struct after = value.getStruct(Envelope.FieldName.AFTER);
+
+        // '517:51:04.777'
+        Long c1 = after.getInt64("c1");
+        Duration c1Time = Duration.ofNanos(c1 * 1_000);
+        assertThat(c1Time.toHours()).isEqualTo(517);
+        assertThat(c1Time.toMinutes()).isEqualTo(31071);
+        assertThat(c1Time.getSeconds()).isEqualTo(1864264);
+        assertThat(c1Time.getNano()).isEqualTo(780000000);
+        assertThat(c1Time.toNanos()).isEqualTo(1864264780000000L);
+        assertThat(c1Time).isEqualTo(Duration.ofHours(517).plusMinutes(51).plusSeconds(4).plusMillis(780));
+
+        // '-13:14:50'
+        Long c2 = after.getInt64("c2");
+        Duration c2Time = Duration.ofNanos(c2 * 1_000);
+        assertThat(c2Time.toHours()).isEqualTo(-13);
+        assertThat(c2Time.toMinutes()).isEqualTo(-794);
+        assertThat(c2Time.getSeconds()).isEqualTo(-47690);
+        assertThat(c2Time.getNano()).isEqualTo(0);
+        assertThat(c2Time.toNanos()).isEqualTo(-47690000000000L);
+        assertTrue(c2Time.isNegative());
+        assertThat(c2Time).isEqualTo(Duration.ofHours(-13).minusMinutes(14).minusSeconds(50));
+
+        // '-733:00:00.0011'
+        Long c3 = after.getInt64("c3");
+        Duration c3Time = Duration.ofNanos(c3 * 1_000);
+        assertThat(c3Time.toHours()).isEqualTo(-733);
+        assertThat(c3Time.toMinutes()).isEqualTo(-43980);
+        assertThat(c3Time.getSeconds()).isEqualTo(-2638800);
+        assertThat(c3Time.getNano()).isEqualTo(1000000);
+        assertThat(c3Time.toNanos()).isEqualTo(-2638799999000000L);
+        assertTrue(c3Time.isNegative());
+        assertThat(c3Time).isEqualTo(Duration.ofHours(-733).plusMillis(1));
+
+        // '-1:59:59.0011'
+        Long c4 = after.getInt64("c4");
+        Duration c4Time = Duration.ofNanos(c4 * 1_000);
+        assertThat(c4Time.toHours()).isEqualTo(-1);
+        assertThat(c4Time.toMinutes()).isEqualTo(-119);
+        assertThat(c4Time.getSeconds()).isEqualTo(-7199);
+        assertThat(c4Time.getNano()).isEqualTo(1000000);
+        assertThat(c4Time.toNanos()).isEqualTo(-7198999000000L);
+        assertTrue(c4Time.isNegative());
+        assertThat(c4Time).isEqualTo(Duration.ofHours(-1).minusMinutes(59).minusSeconds(59).plusMillis(1));
+
+        // '-838:59:58.999999'
+        Long c5 = after.getInt64("c5");
+        Duration c5Time = Duration.ofNanos(c5 * 1_000);
+        assertThat(c5Time.toHours()).isEqualTo(-838);
+        assertThat(c5Time.toMinutes()).isEqualTo(-50339);
+        assertThat(c5Time.getSeconds()).isEqualTo(-3020398);
+        assertThat(c5Time.getNano()).isEqualTo(999999000);
+        assertThat(c5Time.toNanos()).isEqualTo(-3020397000001000L);
+        assertTrue(c5Time.isNegative());
+        assertThat(c5Time).isEqualTo(Duration.ofHours(-838).minusMinutes(59).minusSeconds(58).plusNanos(999999000));
```

Comment:
@rk3rn3r, I've been thinking a bit more about the nano question and I think this assertion isn't correct.

Instead, it should be `minusNanos()` at the end. The reason being that a negative `TIME` value in MySQL such as `-00:00:01.500` should be interpreted as "-1.5 seconds", not as "-1 second plus 0.5 seconds" (i.e. "-0.5 seconds"). This is an example I ran in MySQL to see how it behaves:

```
select mytime, duration, addtime(mytime, duration) from customers where id  =1001;
+---------------------+---------------+---------------------------+
| mytime              | duration      | addtime(mytime, duration) |
+---------------------+---------------+---------------------------+
| 2017-11-28 12:59:40 | -00:00:01.500 | 2017-11-28 12:59:38.500   |
+---------------------+---------------+---------------------------+
```

Note how the result of the calculation is 1.5 seconds less than the original value, not -1 + 0.5 seconds.

Based on that, I think the assertions as well as the implementations should be adjusted. WDYT?

Reply:
@gunnarmorling you're right for the mysql part, but when I compared the inner nanoseconds value representation of the java.time.Duration (call .toNanos()), then the correct value cane only when I added not substracted the nanos. let me double-check this tomorrow.